presto-docs/src/main/sphinx/connector/hive.rst

```diff
@@ -410,6 +407,333 @@ Drop a schema::
+  that result from parallelization. The optimal value should be aligned with the average query
+  size in the system.
+
+
+``hive.metastore.partition-batch-size.max``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Integer`` (at least ``1``)
+ * **Default value:** ``100``
+ * **Description:**
+
+  This together with ``hive.metastore.partition-batch-size.min`` defines the range of partition
+  sizes read from Hive. The first partition is always of size ``hive.metastore.partition-batch-size.min``
+  and each following partition is two times bigger than previous up to
+  ``hive.mestastore.partition-batch-size.max`` (the formula for partition size ``n`` is
+  min(``hive.metastore.partition-batch-size.max``, (``2``^``n``) * ``hive.metastore.partition-batch-size.min``)).
+  This algorithm allows for live adjustment of partition size according to the processing requirements.
+  If the queries in the system will differ significantly from each other in size, then this range should be
+  extended to better adjust to processing requirements. If the queries in the system will mostly be of the
+  same size, then setting both values to the same maximally tuned value may give a slight edge in
+  processing time.
+
+
+``hive.metastore.partition-batch-size.min``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Integer`` (at least ``1``)
+ * **Default value:** ``10``
+ * **Description:**
+
+  See ``hive.metastore.partition-batch-size.max``.
+
+
+``hive.orc.max-buffer-size``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (data size)
+ * **Default value:** ``8 MB``
+ * **Description:**
+
+  Serves as default value for ``orc_max_buffer_size`` session properties defining max size
+  of ORC read operators. Higher value will allow bigger chunks to be processed but will
+  decrease concurrency level.
+
+
+``hive.orc.max-merge-distance``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (data size)
+ * **Default value:** ``1 MB``
+ * **Description:**
+
+  Serves as the default value for the ``orc_max_merge_distance`` session property. Two reads
+  from an ORC file may be merged into a single read if the distance between the requested data
+  ranges in the data source is less than or equal to this value.
+
+
+``hive.orc.stream-buffer-size``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (data size)
+ * **Default value:** ``8 MB``
+ * **Description:**
+
+  Serves as the default value for the ``orc_max_buffer_size`` session property. It defines the
+  maximum size of ORC read operators. A higher value will allow bigger chunks to be processed,
+  but will decrease concurrency.
+
+
+``hive.orc.use-column-names``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Boolean``
+ * **Default value:** ``false``
+ * **Description:**
+
+  Access ORC columns using names from the file. By default, Hive access columns in ORC files
+  using the order recoded in the Hive metastore. Setting this property allows to use columns
```

Comment:
what does the `order` refer to? Is refers to order of how columns (column names) are stored in a file? Is there any impact on query performance? If so, when and why?

Reply:
changed the description a bit..there is no perf impact for this change